cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,67 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+
+const faker = require('faker');
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
+  let user;
+
+  const modalMessage = {
+    general: 'Registration failed!',
+    emptyPassword: 'Password field required.',
+    takenEmail: 'Email already taken.',
+    successRegistration: 'Your registration was successful!',
+  };
+
+  const user2 = {
+    username: faker.name.firstName(),
+    email: faker.internet.email(),
+    password: faker.internet.password(),
+  };
+
   before(() => {
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
+  });
 
+  beforeEach(() => {
+    cy.task('db:clear');
   });
 
-  it('should ...', () => {
+  it('should provide an ability to sign up with user credentials', () => {
+    signUpPage.visit();
+    signUpPage.fillUsernameField(user.username);
+    signUpPage.fillEmailField(user.email);
+    signUpPage.fillPasswordField(user.password);
+    signUpPage.clickOnSignUpBtn();
+
+    homePage.assertRegistrationSuccessModal(modalMessage.successRegistration);
+    homePage.clickOnModalOkBtn();
+    homePage.assertUsername(user.username);
+  });
+  it('should not provide an ability to sign up with taken email', () => {
+    cy.register(user.email, user.username, user.password);
+
+    signUpPage.visit();
+    signUpPage.fillUsernameField(user2.username);
+    signUpPage.fillEmailField(user.email);
+    signUpPage.fillPasswordField(user2.password);
```

Comment:
You are trying to sign up with 'user2.password', but you never set it before. As a result, the password could be empty during the test run, leading to a failed test.